.github/workflows/tests.yml

```diff
@@ -0,0 +1,77 @@
+# This workflow runs test
+
+name: tests
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "develop"
+  push:
+    branches:
+      - "main"
+      - "develop"
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        py-version:
+          - 3.6
+          - 3.7
+          - 3.8
+          - 3.x
+        mpi:
+          - openmpi
+        os:
+          - ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install MPI (${{ matrix.mpi }})
+        run: bash .bash/install-${{ matrix.mpi }}.sh
+
+      - name: Set up Python ${{ matrix.py-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.py-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install lyncs_setuptools
+
+      - name: Install via pip
+        run: |
+          sudo pip install -e .[all]
+
+      - name: Run serial tests
+        run: |
+          coverage run --rcfile=setup.cfg -m pytest
+
+      - name: Run parallel tests (procs=2, 4, 7, 12)
+        run: |
+          PROCS=(2 4 7 12)
+          for proc in "${PROCS[@]}"
+          do
+            echo "::group::Run parallel tests (procs=$proc)"
+            mpirun -n $proc coverage run --rcfile=setup.cfg -m pytest --with-mpi
+            echo "::endgroup::"
+          done
+
+      - name: Coverage report
+        run: |
+          coverage combine
+          coverage report
```

Comment:
@cstyl 
FYI now we get a coverage report here and in the next step the coverage is uploaded on codecov.io only if the master branch